### PR TITLE
Handle ps container created field as a time.Time

### PR DESCRIPTION
**Pull request by rhatdan**
In the current code we were translating the created time
from a time.Time to a unix epoch, this was leading to a loss
of precession, and some unexpected results where the sorting
order of containers was misordered because of the precession loss.

Fixes: https://github.com/containers/podman/issues/8414

If we pass around created as time.Time, we do not loose the precission.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
@edsantiago PTAL

File: pkg/ps/ps.go
Comment (TomSweeneyRedHat) [inline pkg/ps/ps.go:304]: I'm a bit confused.  Was the ">" correct for the original version of this "Less()" function?  If so, does this need to be `After` rather than `Before`, or are you correcting a bad sort?

Back to my tea cup.

File: pkg/domain/entities/container_ps.go
Comment (mheon) [inline pkg/domain/entities/container_ps.go:18]: @rhatdan @jwhonce Is this going over the wire? As in, is this a breaking change to our JSON format?

File: pkg/ps/ps.go
Comment (jwhonce) [reply pkg/ps/ps.go:304]: @TomSweeneyRedHat I'm a mistake in original code see https://github.com/containers/podman/pull/8427/files#diff-6fb97e57a192c9eae49da7e6a2ca7783e8d30eaf4d323cf55b56b28861016eb4L140

**Review by jwhonce — APPROVED** [review]
LGTM

File: pkg/ps/ps.go
Comment (TomSweeneyRedHat) [reply pkg/ps/ps.go:304]: @jwhonce "I'm a mistake in original code", I wasn't aware you've gone full Tron in this project.  That's a bit scary! :smile: 
Thx for the 411

**Comment by TomSweeneyRedHat** [comment]
LGTM
as long as @jwhonce is comfortable with the date type change.

File: pkg/domain/entities/container_ps.go
Comment (rhatdan) [reply pkg/domain/entities/container_ps.go:18]: I don't believe this is going over the wire.  The time.Time came over the wire though.

Comment (rhatdan) [reply pkg/domain/entities/container_ps.go:18]: But this may be a breaking change for anyone who has vendored in /pkg/ directories. So we might have to wait to fix this until 3.0.

**Comment by rhatdan** [comment]
@vrothberg  Do you agree this is a breaking change, wait for 3.0>

**Comment by edsantiago** [comment]
Well, the good news is, in one hour of loop testing I got no failures. LGTM.

BTW I have a test for it in the works, will submit PR some time later, possibly with a `FIXME 8427` comment if this can't merge yet.

**Comment by rhatdan** [comment]
We are holding up the merge of this PR until after 2.2 release.

**Comment by rhatdan** [comment]
@containers/podman-maintainers PTAL This is ready to merge.

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Pull request merged**
